tools/cross_compile.sh

```diff
@@ -265,6 +265,25 @@ QEMU_ARGS+=( -L "${SYSROOT_DIR}" )
 QEMU_ARGS+=( -E LD_PRELOAD="${SYSROOT_DIR}/usr/lib/libstdc++.so.6:${SYSROOT_DIR}/lib/libgcc_s.so.1" )
 }
 
+function expand_wasm_config() {
+  local -r EMSDK_VERSION=2.0.14
```

Comment:
Last Tag as of 2024 June 07
https://github.com/emscripten-core/emsdk/releases/tag/3.1.61

Reply:
Hi Mizux,
Is it planned for next release or will it take time? Also, is this command work with stable branch:
emcmake cmake -S . -B build -DBUILD_DEPS=ON -DBUILD_PYTHON=ON

I tried after making changes in cross_compile.sh and I am getting error as:
Could NOT find Python3 (missing: Python3_INCLUDE_DIRS Python3_LIBRARIES Development Development.Module Development.Embed) (found version "3.11.9")

However, I don't see such error when using just cmake?

Reply:
Sorry, I don't get it, what did you try to do/expect ?

`BUILD_PYTHON` is used to generate automatically python wrapper on top of the C++ library to have access to ortools in python (which is just a wrapper on top of the native library, since we won't reimplement ortools in pure Python/Java nor .Net).

On the other side, AFAIK, `emcmake` is used to transpile C++ code into wasm byte code to have a "browser compatible" source code so there is no point to try to build a python wrapper on top of this wasm library.

Reply:
Hi there, I recently pointed @airen1986 to this issue from https://github.com/pyodide/pyodide/issues/5013. If configured correctly, Pyodide would allow runtime for a Python interface for this WASM library through an Emscripten-compiled Python extension module.

Reply:
ok, I guess I understand what you want to do:
0. This is not something we support
1. currently our wrapper is generated by swig which is currently use to build 64bit code -> don't know how it work with WASM32 ptr size etc...
2. IIRC emcmake is working by injecting a CMake cross toolchain thus my bet is the Toolchainfile do not provide a python thus the issue.
3. I would suggest to play with my pet project: https://github.com/Mizux/python-native which should have the same cmake stuff than or-tools so once you have something work there it should be easier to patch/fixup or-tools and dev iteration should be faster IMHO...

sorry this comment is little bit dry, will try to reformulate if you want...